### ISSUE TITLE
Folder appears empty when >999 child files/folders

### ISSUE THREAD

**Issue by dizzystuff** [silverstripe/silverstripe-asset-admin#1269]
Using: SS 4.10, GraphQL 4/dev-master

I have an folder which contains ~1700 assets. Recently when browsing to this folder via asset admin, it appears empty. No errors in UI or console, just "No items found" message.

Plenty of testing and debugging later, and it appears that this occurs once the folder contains greater than 999 items (files/folders). I can replicate this is fresh folders/files reliably.

Unfort I haven't yet been able to figure out when whatever is causing this was introduced, it was only recently reported by the user.

Is this a new and/or undocumented limitation on asset admin? Or is it a bug? My testing suggests it shouldn't only be me that's experiencing it, but if you think that's the case I'll go back to the drawing board. 

I figure there's a chance it's related to some kind of GraphQL constraint or the instability with GQL v4 that is beyond me, but the performance improvements in v4 simply make it a practical necessity to use it on this project (and if that's heading into beta soon, and this is related, it's worth finding a resolution for).

### ACs
- [ ] Investigate if this bug is replicatable with graphql 3 or if it is graphql 4 only
- [ ] Determine if this bug is graphql3 related or if it is asset-admin related (probably javascript if it is)
- [ ] Folders with over 999 files render correctly with graphql 3 and graphql 4

**Comment by sabina-talipova** [comment]
YA ne smogla vosproizvesti dannuyu problemu v moyey lokal'noy srede. YA sozdala neskol'ko papok. V odnoy iz nikh nakhoditsya boleye 1000 papok. V drugoy nakhoditsya boleye 3599 faylov v formate txt . Iznachal'no, ya ispol'zovala defoltnyye znacheniya dlya kolichestva elementov na stranitse. Fayly byli korrektno raspredeleny na kazhdoy stranitse po 50 elementov na stranitse. Daleye ya sozdala konfiguratsionnyy fayl i ustanovila novoye znacheniye dlya kolichestva elementov na stranitse. # SilverStripe\AssetAdmin\Controller\AssetAdmin: # page_length: 4000 # max_history_entries: 10000# Posle etogo voznoikla raskhozhdeniye mezhdu znacheniyem page_lengtkh i kolichestvom elementov na stranitse. Ne smotrya na to chto znacheniye bylo ustanovleno 4000 elementov na stranitse, ya ne poluchala boleye 100 elementov v zaprose GrafKL. No ya ispravila dannuyu problemu putem dobavleniya dopolnitel'noy konfiguratsii # SilverStripe\GraphQL\Schema\Plugin\PaginationPlugin: # default_limit: 10000 # max_limit: 10000 Posle etogo vse rabotayet korrektno i ya vizhu na stranitse vse 3599 faylov.
Show more
1,049 / 5,000
Translation results
I was unable to reproduce this issue in my local environment. I have created several folders. One of them contains more than 1000 folders. The other contains over 3599 files in txt format.
Initially, I used default values ​​for the number of elements per page. The files have been correctly distributed on each page with 50 items per page. Next, I created a config file and set a new value for the number of items per page.
```yml
SilverStripe\AssetAdmin\Controller\AssetAdmin:
   page_length: 4000
   max_history_entries: 10000
```

After that, there was a discrepancy between the value of page_langth and the number of elements on the page. Even though the value was set to 4000 items per page, I didn't get more than 100 items in the GrafKL query.
But I fixed this problem by adding additional configuration
``yml
SilverStripe\GraphQL\Schema\Plugin\PaginationPlugin:
   default_limit: 10000
   max_limit: 10000
```
After that, everything works correctly and I see all 3599 files on the page.

**Comment by sabina-talipova** [comment]
@dizzystuff, probably, you could provide us with more details about test steps that you've done to replicate this issue. 
Thank you.

**Comment by sabina-talipova** [comment]
CLOSED. 
Since I was unable to reproduce this issue locally, I am closing this issue. Perhaps it was solved with another task.

**Comment by dizzystuff** [comment]
Hi @sabina-talipova sorry it took me a couple of weeks to respond, thank you for looking into this.

Today I resolved the issue, and want to note the details in case anyone else runs across it.

It's the strangest thing. It seems to have been a database-level issue? When getting the children/contents of the Folder, the `File::get()->byIDs($ids)` was doing its job of asking the database for records, but the db seemed to be returning zero records. But then more oddly when I ran the same sql query on the database manually, it worked fine.

That was occuring on MariaDB 10.6.7. 

After an upgrade to MariaDB 10.6.8 (and no other code/environment changes), everything began working as expected, the `File::get()->byIDs($ids)` call is returning all Files with IDs in $ids, and thus the asset admin looks/works as it should.

I developed a hunch during debugging that for some reason the database might have been choking on so many values being in `where File.ID IN (---list of ids--)`. Weird, and weirder still that it was fine when run on the db manually, so I don't know 🤷‍♂️
